blockstore/blockstore.go

```diff
@@ -30,6 +30,11 @@ type BatchDeleter interface {
 	DeleteMany(cids []cid.Cid) error
 }
 
+// BlockstoreIterator is a trait for efficient iteration
+type BlockstoreIterator interface {
+	ForEachKey(func(cid.Cid) error) error
```

Comment:
Probably like 2x?

Reply:
Ah, probably because you don't have a buffer on that channel which forces context switches. They're faster in go, but not _that_ fast.

I'm not really _against_ this version, but internal iteration has some... issues as we've noticed.

Reply:
See https://github.com/filecoin-project/lotus/commit/909f7039d47c351c2581de0c39bb1c9b5d8a721d -- finally solved our issues :)